// Generated from ifex_dispatcher.ifex by covesa-ifex-core
// DO NOT EDIT - regenerate with: ./generate_proto.sh
syntax = "proto3";



package swdv.ifex_dispatcher;

// Namespace: core
// Core dynamic method calling functionality

// Method call execution status
enum call_status_t {
  SUCCESS = 0;
  FAILED = 1;
  TIMEOUT = 2;
  SERVICE_UNAVAILABLE = 3;
  METHOD_NOT_FOUND = 4;
  INVALID_PARAMETERS = 5;
}

// Method call request information
message method_call_t {
  string service_name = 1;
  string method_name = 2;
  string parameters = 3;
  uint32 timeout_ms = 4;
}

// Method call execution result
message call_result_t {
  call_status_t status = 1;
  string response = 2;
  string error_message = 3;
  uint32 duration_ms = 4;
  string service_endpoint = 5;
}

// Method: call_method
// Execute a method on any IFEX service
message call_method_request {
  method_call_t call = 1;
}

message call_method_response {
  call_result_t result = 1;
}

service call_method_service {
  rpc call_method(call_method_request) returns (call_method_response);
}


